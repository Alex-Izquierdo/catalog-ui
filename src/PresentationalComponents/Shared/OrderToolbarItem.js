import React from 'react';
import { Section } from '@red-hat-insights/insights-frontend-components';
import { ToolbarGroup, ToolbarItem, Dropdown, DropdownPosition,
    DropdownToggle, DropdownItem, KebabToggle, Title, Button } from '@patternfly/react-core';
import { css } from '@patternfly/react-styles';
import spacingStyles from '@patternfly/patternfly-next/utilities/Spacing/spacing.css';
import '../../SmartComponents/Portfolio/portfolio.scss';

export default function OrderToolbarItem(props) {
    return (
        <ToolbarGroup className={ 'pf-u-ml-auto-on-xl' }>
            <ToolbarItem className={ css(spacingStyles.mrXL) }>
                <Button variant="link" id="ordersButton">
                    <i className="fas fa-shopping-cart" aria-hidden="true"></i>
                    Orders
                </Button>
            </ToolbarItem>
        </ToolbarGroup>
    );
};

