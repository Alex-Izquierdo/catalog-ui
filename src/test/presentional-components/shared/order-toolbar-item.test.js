import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import OrderToolbarItem from '../../../PresentationalComponents/Shared/OrderToolbarItem';

describe('<OrderToolbarItem />', () => {
  let initialProps;
  beforeEach(() => {
    initialProps = {};
  });

  it('should render correcntly', () => {
    const wrapper = mount(<MemoryRouter><OrderToolbarItem { ...initialProps } /></MemoryRouter>).find(OrderToolbarItem);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correcntly when required', () => {
    const wrapper = mount(<MemoryRouter><OrderToolbarItem { ...initialProps } /></MemoryRouter>).find(OrderToolbarItem);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
