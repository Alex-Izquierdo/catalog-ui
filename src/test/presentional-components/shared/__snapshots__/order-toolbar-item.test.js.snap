// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderToolbarItem /> should render correcntly 1`] = `
<OrderToolbarItem>
  <ToolbarGroup
    className="pf-u-ml-auto-on-xl"
    style={
      Object {
        "alignSelf": "baseline",
      }
    }
  >
    <div
      className="pf-l-toolbar__group pf-u-ml-auto-on-xl"
      style={
        Object {
          "alignSelf": "baseline",
        }
      }
    >
      <ToolbarItem
        className={null}
      >
        <div
          className="pf-l-toolbar__item"
        >
          <Link
            key="show_order-list"
            replace={false}
            to="/orders"
          >
            <a
              href="/orders"
              onClick={[Function]}
            >
              <Button
                aria-label={null}
                className=""
                component="button"
                id="ordersButton"
                isActive={false}
                isBlock={false}
                isDisabled={false}
                isFocus={false}
                isHover={false}
                type="button"
                variant="link"
              >
                <button
                  aria-disabled={null}
                  aria-label={null}
                  className="pf-c-button pf-m-link"
                  disabled={false}
                  id="ordersButton"
                  tabIndex={null}
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="fas fa-shopping-cart"
                  />
                  Orders
                </button>
              </Button>
            </a>
          </Link>
        </div>
      </ToolbarItem>
    </div>
  </ToolbarGroup>
</OrderToolbarItem>
`;

exports[`<OrderToolbarItem /> should render correcntly when required 1`] = `
<OrderToolbarItem>
  <ToolbarGroup
    className="pf-u-ml-auto-on-xl"
    style={
      Object {
        "alignSelf": "baseline",
      }
    }
  >
    <div
      className="pf-l-toolbar__group pf-u-ml-auto-on-xl"
      style={
        Object {
          "alignSelf": "baseline",
        }
      }
    >
      <ToolbarItem
        className={null}
      >
        <div
          className="pf-l-toolbar__item"
        >
          <Link
            key="show_order-list"
            replace={false}
            to="/orders"
          >
            <a
              href="/orders"
              onClick={[Function]}
            >
              <Button
                aria-label={null}
                className=""
                component="button"
                id="ordersButton"
                isActive={false}
                isBlock={false}
                isDisabled={false}
                isFocus={false}
                isHover={false}
                type="button"
                variant="link"
              >
                <button
                  aria-disabled={null}
                  aria-label={null}
                  className="pf-c-button pf-m-link"
                  disabled={false}
                  id="ordersButton"
                  tabIndex={null}
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="fas fa-shopping-cart"
                  />
                  Orders
                </button>
              </Button>
            </a>
          </Link>
        </div>
      </ToolbarItem>
    </div>
  </ToolbarGroup>
</OrderToolbarItem>
`;
